File: imp/coletor_gps/Status.java
package coletor_gps;

class Status
{
	public enum Semaforo
	{
		Verde(0),
		Amarelo(1),
		Vermelho(2);
		
		private final int codigoSemaforo;
		
		Semaforo(int codigoSemaforo)
		{
			this.codigoSemaforo = codigoSemaforo;
		}

		public int getCodigoSemaforo()
		{
			return (codigoSemaforo);
		}
	};

	private GerenteStatus gerenteStatus;
	private Semaforo semaforo;

	public Status(GerenteStatus gerenteStatus)
	{
		this.gerenteStatus = gerenteStatus;
	}

	public GerenteStatus getGerenteStatus()
	{
		return (gerenteStatus);
	}

	public Semaforo getSemaforo()
	{
		return (semaforo);
	}

	public void setSemaforo(Status.Semaforo semaforo)
	{
		this.semaforo = semaforo;
	}
	
	public void notificarGerente(IStatusProdutor produtor)
	{
		gerenteStatus.atualizarStatusGlobal(produtor);
	}
}

File: imp/Localizador/Status.cpp
#include "Status.h"

Status::Status(GerenteStatus *gerenteStatus) {
        this->gerenteStatus = gerenteStatus;
        semaforo = Semaforo::NORMAL;
        pinMode(5, OUTPUT);
        pinMode(6, OUTPUT);
        pinMode(7, OUTPUT);
        limpar();
}

void
Status::limpar(void) {
        digitalWrite(5, LOW);
        digitalWrite(6, LOW);
        digitalWrite(7, LOW);
}

Semaforo
Status::getSemaforo(void) {
        return (semaforo);
}

void
Status::setSemaforo(Semaforo semaforo) {
        this->semaforo = semaforo;
}
